### PR TITLE
Fix sbt console

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes sbt/sbt#3129

### CODE REVIEW

**Review by eed3si9n — APPROVED** [review]
This brings back the old console API to unblock sbt.
LGTM

**Review by jvican — APPROVED** [review]
LGTM.

File: internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala [hunk 1]
```diff
@@ -81,17 +76,8 @@ final class AnalyzingCompiler(
...
-      changes,
-      arguments.toArray,
-      output,
-      callback,
-      reporter,
-      cache,
-      log,
-      Maybe.nothing[CompileProgress]
-    )
+    val progress = Maybe.nothing[CompileProgress]
```
Comment (jvican) [inline internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala:79]: :smile:

**Review by jvican — COMMENTED** [review]
I'll merge when CI passes.

**Comment by jvican** [comment]
Weird, CI is not running at all.

**Comment by jvican** [comment]
Unfortunately, merging the formatting PR made this PR have conflicts. Can you update @dwijnand and I merge?

**Comment by dwijnand** [comment]
Yeah, I knew merging your PR would have this effect. But I did it because it was 9 days in the making and this one is brand new, so I'll rebase my work.

**Pull request merged**